src/java.desktop/macosx/native/libawt_lwawt/awt/AWTEvent.m

```diff
@@ -460,19 +460,35 @@ static unichar NsGetDeadKeyChar(unsigned short keyCode)
     }
 
     if ([[NSCharacterSet letterCharacterSet] characterIsMember:ch]) {
+        TISInputSourceRef source = TISCopyCurrentKeyboardInputSource();
+        NSString* lang;
+        NSArray* languages = (NSArray *) TISGetInputSourceProperty(source, kTISPropertyInputSourceLanguages);
+        if (languages && [languages count] > 0) {
+            lang = [languages objectAtIndex:0];
+        } else {
+            lang = languages;
+        }
+
```

Comment:
This may become irrelevant if the fix changes, as it well might, but I'm not sure what to make of line 469 above.

the check "If (languages .." implies that there's a chance languages may be null  and I'm a bit
surprised the compiler doesn't complain about the assignment of an NSArray* to an NSString*
and then later on you dereference lang (line 483)